File: 206 Reverse Linked List.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun reverseList(head: ListNode?): ListNode? {
        var current = head
        var prev: ListNode? = null
        while (current != null) {
            val temp = current.next
            current.next = prev
            prev = current
            current = temp
        }
    return prev
    }
}
// Runtime: 152 ms, faster than 95.13% of Kotlin online submissions for Reverse Linked List.
// Memory Usage: 36.2 MB, less than 27.49% of Kotlin online submissions for Reverse Linked List.

File: 206 Reverse Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *current = head; 
    struct ListNode *prev = NULL;  
    struct ListNode *next = NULL;  

    while(current!=NULL) {
        next = current->next; 
        current->next = prev; 
        prev = current;
        current = next;
    }
    head = prev;
    return head;
}


Runtime: 0 ms, faster than 100.00% of C online submissions for Reverse Linked List.
Memory Usage: 6.4 MB, less than 68.89% of C online submissions for Reverse Linked List.
